client/client.go

```diff
@@ -281,21 +280,131 @@ func newHttpClient() *http.Client {
 	return &http.Client{}
 }
 
-func dialOpenSSL(conf Config, network string, addr string) (net.Conn, error) {
-	contextSSL, err := openssl.NewCtx()
-	// probably should consider reusing the context, but then we
-	// have to propagate it with every request.
-	err = contextSSL.LoadVerifyLocations(conf.ServerCert, "")
+func listSystemCertsFound(certDir string) (int, error) {
+	sysCertsFound := 0
+	files, err := ioutil.ReadDir(certDir)
 	if err != nil {
-		return nil, err
+		return 0, fmt.Errorf("Failed to read the OpenSSL default directory (%s). Err %v", certDir, err.Error())
 	}
+	for _, certFile := range files {
+		certBytes, err := ioutil.ReadFile(path.Join(certDir, certFile.Name()))
+		if err != nil {
+			log.Debugf("Failed to read the certificate file for the HttpsClient. Err %v", err.Error())
+			continue
+		}
+
+		certs := openssl.SplitPEM(certBytes)
+		if len(certs) == 0 {
+			log.Debugf("No PEM certificate found in '%s'", certFile)
+			continue
+		}
+		first, certs := certs[0], certs[1:]
+		_, err = openssl.LoadCertificateFromPEM(first)
+		if err != nil {
+			log.Debug(err.Error())
+		} else {
+			sysCertsFound += 1
+		}
+	}
+	return sysCertsFound, nil
+}
+
+func loadServerTrust(ctx *openssl.Ctx, conf *Config) (*openssl.Ctx, error) {
+	defaultCertDir, err := openssl.GetDefaultCertificateDirectory()
+	if err != nil {
+		return ctx, errors.New("Failed to get the default OpenSSL certificate directory. Please verify the OpenSSL setup")
+	}
+	sysCertsFound, err := listSystemCertsFound(defaultCertDir)
+	if err != nil {
+		log.Warnf("Failed to list the system certificates with error: %v", err)
+	}
+
+	err = ctx.SetDefaultVerifyPaths()
+	if err != nil {
+		return ctx, fmt.Errorf("Failed to set the default OpenSSL directory. OpenSSL error code: %s", err.Error())
+	}
+	err = ctx.LoadVerifyLocations(conf.ServerCert, "")
+	if err != nil {
+		if strings.Contains(err.Error(), "No such file or directory") {
+
+			log.Errorf(errMissingServerCertF, conf.ServerCert)
+		} else {
+			log.Errorf("Failed to Load the Server certificate. Err %v", err.Error())
+		}
+		if sysCertsFound == 0 {
+			log.Error(errMissingCerts)
```

Comment:
And therefore we can once again issue the `errMissingCerts` error, like you requested, if no certificates were loaded - neither configuration server cert, or system certificates.

Reply:
:+1: